lib/online_order.rb

```diff
@@ -0,0 +1,56 @@
+require 'csv'
+require_relative '../lib/costumer'
+require_relative '../lib/order'
+
+module Grocery
+  class OnlineOrder < Grocery::Order
+    attr_reader :costumer, :status, :id, :products
+
+    def initialize(id, products, costumer, status) #= :pending)
+      @costumer = costumer
+      @status = status
+      #I have to initialize the products??
+      @products = products
```

Comment:
Instead of saving `products` yourself here, you should be calling `super`, which will run the constructer for `Order`. Since that takes two parameters, you'll have to pass them in - it'll look like this:

```ruby
def initialize(id, products, costumer, status) #= :pending)
  @costumer = costumer
  @status = status
  super(id, products)
end
```